src/server/editorServices.ts

```diff
@@ -1217,13 +1217,35 @@ namespace ts.server {
             }
             else {
                 const project = this.createProject(configFilename, projectOptions);
+                let programSize = 0;
+
+                // As the project openning might not be complete if there are too many files,
+                // therefore to surface the diagnostics we need to make sure the given client file is opened.
+                if (clientFileName) {
+                    const currentClientFileInfo = this.openFile(clientFileName, /*openedByClient*/ true);

```

Comment:
Do you not need to worry about the `this.host.fileExists` check that used to be performed for all files? (i.e. is the clientFileName guaranteed to exit)?